### PR TITLE
Forward logs to Sentry as breadcrumbs

### PR DESCRIPTION
**Pull request by saket**
image

### CODE REVIEW

**Comment by saket** [comment]
@vinaysshenoy @pratul should we also log network calls to Timber? Just the endpoint and the response code.

**Comment by vinaysshenoy** [comment]
> @vinaysshenoy @pratul should we also log network calls to Timber? Just the endpoint and the response code.

Hmm. Might be useful. But some endpoints contain user IDs and phone numbers. Not sure if those should go to Sentry.

File: app/src/main/java/org/simple/clinic/crash/CrashBreadcrumbsTimberTree.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  override fun log(priority: Int, tag: String?, message: String, error: Throwable?) {
+    val priorityEnum = when (priority) {
+      Log.VERBOSE -> Breadcrumb.Priority.VERBOSE
+      Log.DEBUG -> Breadcrumb.Priority.DEBUG
+      Log.INFO -> Breadcrumb.Priority.INFO
+      Log.WARN -> Breadcrumb.Priority.WARN
+      Log.ERROR -> Breadcrumb.Priority.ERROR
+      Log.ASSERT -> Breadcrumb.Priority.ASSERT
+      else -> throw AssertionError()
```
[the remark this answers is not in the document]
Comment (saket) [reply app/src/main/java/org/simple/clinic/crash/CrashBreadcrumbsTimberTree.kt]: Done

File: app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt [hunk 2]
```diff
@@ -3,10 +3,39 @@ package org.simple.clinic.crash
...
+    val sentryBreadcrumb = BreadcrumbBuilder()
+        .setLevel(priorityToLevelString(breadcrumb.priority))
+        .setCategory(breadcrumb.tag)
+        .setMessage(breadcrumb.message)
+        .setTimestamp(Date(System.currentTimeMillis()))
+        .build()
+    Sentry.getContext().recordBreadcrumb(sentryBreadcrumb)
+  }
+
+  private fun priorityToLevelString(priority: Breadcrumb.Priority): SentryBreadcrumbLevel {
```
[the remark this answers is not in the document]
Comment (saket) [reply app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt]: Done

File: app/src/main/java/org/simple/clinic/crash/CrashBreadcrumbsTimberTree.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package org.simple.clinic.crash
+
+import android.util.Log
+import timber.log.Timber
+
+class CrashBreadcrumbsTimberTree(private val crashReporter: CrashReporter) : Timber.Tree() {
+
+  override fun log(priority: Int, tag: String?, message: String, error: Throwable?) {
+    val priorityEnum = when (priority) {
+      Log.VERBOSE -> Breadcrumb.Priority.VERBOSE
```
[the remark this answers is not in the document]
Comment (saket) [reply app/src/main/java/org/simple/clinic/crash/CrashBreadcrumbsTimberTree.kt]: Done

**Pull request merged**